samples/subsys/fs/littlefs/src/main.c

```diff
@@ -12,11 +13,249 @@
 #include <device.h>
 #include <fs/fs.h>
 #include <fs/littlefs.h>
+#include <logging/log.h>
 #include <storage/flash_map.h>
 
+LOG_MODULE_REGISTER(main);
+
 /* Matches LFS_NAME_MAX */
 #define MAX_PATH_LEN 255
+#define TEST_FILE_SIZE 547
+
+static uint8_t file_test_pattern[TEST_FILE_SIZE];
+static int lsdir(const char *path)
+{
+	int res;
+	struct fs_dir_t dirp;
+	static struct fs_dirent entry;
+
+	fs_dir_t_init(&dirp);
+
+	/* Verify fs_opendir() */
+	res = fs_opendir(&dirp, path);
+	if (res) {
+		LOG_ERR("Error opening dir %s [%d]\n", path, res);
+		return res;
+	}
+
+	LOG_PRINTK("\nListing dir %s ...\n", path);
+	for (;;) {
+		/* Verify fs_readdir() */
+		res = fs_readdir(&dirp, &entry);
+
+		/* entry.name[0] == 0 means end-of-dir */
+		if (res || entry.name[0] == 0) {
+			break;
+		}
+
+		if (entry.type == FS_DIR_ENTRY_DIR) {
+			LOG_PRINTK("[DIR ] %s\n", entry.name);
+		} else {
+			LOG_PRINTK("[FILE] %s (size = %zu)\n",
+				   entry.name, entry.size);
+		}
+	}
+
+	/* Verify fs_closedir() */
+	fs_closedir(&dirp);
+
+	return res;
+}
+
+static int littlefs_increase_infile_value(char *fname)
+{
+	uint8_t boot_count = 0;
+	struct fs_file_t file;
+	int rc;
+
+	fs_file_t_init(&file);
+	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
+	if (rc < 0) {
+		LOG_ERR("FAIL: open %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_read(&file, &boot_count, sizeof(boot_count));
+	if (rc < 0) {
+		LOG_ERR("FAIL: read %s: [rd:%d]", log_strdup(fname), rc);
+		goto out;
+	}
+	LOG_PRINTK("%s read count:%u (bytes: %d)\n", fname, boot_count, rc);
+
+	rc = fs_seek(&file, 0, FS_SEEK_SET);
+	if (rc < 0) {
+		LOG_ERR("FAIL: seek %s: %d", log_strdup(fname), rc);
+		goto out;
+	}
+
+	boot_count += 1;
+	rc = fs_write(&file, &boot_count, sizeof(boot_count));
+	if (rc < 0) {
+		LOG_ERR("FAIL: write %s: %d", log_strdup(fname), rc);
+		goto out;
+	}
+
+	LOG_PRINTK("%s write new boot count %u: [wr:%d]\n", fname,
+		   boot_count, rc);
+
+ out:
+	rc = fs_close(&file);
```

Comment:
The `rc` gets overwritten here, so in case when we get to `out` due to error and are successfully closing the file, the overall return from function would be success. This does not seem to be correct.